module/localisation/RobotParticleLocalisation/src/RobotParticleLocalisation.cpp

```diff
@@ -101,85 +98,100 @@ namespace localisation {
         on<Trigger<ResetRobotHypotheses>, With<Sensors>, Sync<RobotParticleLocalisation>>().then(
             "Reset Robot Hypotheses",
             [this](const ResetRobotHypotheses& locReset, const Sensors& sensors) {
-                Transform3D Hfw;
-                const Transform3D& Htw = convert(sensors.Htw);
-                std::vector<arma::vec3> states;
-                std::vector<arma::mat33> cov;
+                std::vector<Eigen::Vector3d> states;
+                std::vector<Eigen::Matrix3d> cov;
+
+                Eigen::Affine3d Htw;
+                Htw.matrix() = (sensors.Htw);
 
                 for (auto& s : locReset.hypotheses) {
-                    Transform3D Hft;
-                    arma::vec3 rTFf   = {s.position[0], s.position[1], 0};
+                    const Eigen::Vector3d rTFf(s.position.x(), s.position.y(), 0);
```

Comment:
If `rTFf` is not used anywhere else then just assign this directly to the affine transform.
```
Hft.translation() = {s.position[0], s.position[1], 0};
```